modules/core/04-channel/keeper/packet_test.go

```diff
@@ -1021,3 +1066,75 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 		})
 	}
 }
+
+// TestAcknowledgeFlushStatus tests that Acknowledging the last in-flight packet moves the channel
+// flush status to FLUSHCOMPLETE.
+func (suite *KeeperTestSuite) TestAcknowledgeFlushStatus() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		msg                 string
+		malleate            func()
+		expectedFlushStatus types.FlushStatus
+	}{
+		{
+			"success: final packet commitment cleared, flush status set to FLUSHCOMPLETE",
+			func() {},
+			types.FLUSHCOMPLETE,
+		},
+		{
+			"success: has in-flight packets, flush status remains FLUSHING",
+			func() {
+				// Send an additional packet
+				sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+				suite.Require().Equal(uint64(2), sequence)
+				suite.Require().NoError(err)
+			},
+			types.FLUSHING,
+		},
+	}
+
+	for i, tc := range testCases {
+		tc := tc
+		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.msg, i, len(testCases)), func() {
+			suite.SetupTest() // reset
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			// create packet commitment
+			sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+
+			// create packet receipt and acknowledgement
+			packet := types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
```

Comment:
I think we could also reuse this rather than fully building the expected packet here

```
packet := ibctesting.ParsePacketFromEvents(suite.chainA.GetContext().EventManager().Events())
```

Reply:
curiously, this doesn't work, events are empty, am not seeing why

Reply:
ah, calling `CommitBlock` in the endpoint's `SendPacket` probably flushes the events from the event manager on the current context.

Most certainly easier to leave this as is, forget my suggestion 😅

Reply:
specifically its when we call `NextBlock` in the testing pkg, this will call abci `Commit()` and `BeginBlock()` which will create a new ctx https://github.com/cosmos/cosmos-sdk/blob/v0.47.3/baseapp/abci.go#L184-L193

Reply:
now that's what I call ~~music~~ context :heart: